src/MassTransit.RabbitMqTransport/Topology/RabbitMqExchangeBindingExtensions.cs

```diff
@@ -63,9 +67,9 @@ public static ExchangeBindingSettings GetExchangeBinding(this SendSettings setti
 
         public static IEnumerable<ExchangeBindingSettings> GetExchangeBindings(this ReceiveSettings settings, string exchangeName)
         {
-            var exchange = new Exchange(exchangeName, settings.Durable, settings.AutoDelete, settings.ExchangeType);
-
-            var binding = new ExchangeBinding(exchange);
+            var exchange = new Exchange(exchangeName, settings.Durable, settings.AutoDelete, settings.ExchangeTypeProvider);
+            //TODO find a way for this to stay working while you want to implement routingkey structure

```

Comment:
It indicates uncertainty, on my part, about the impact this function will have on customized routing logic.
Since the current in the function bindings are created without the info (messageType) needed for  the custom routing(key) logic.
I will investigate the impact further (now that this solution is being considered) and find a solution (if the method in it's current state does not play well with the custom routing).